packages/new-editor/src/components/editor/variables/detail/Variable.tsx

```diff
@@ -1,34 +1,49 @@
 import { Fieldset, Input, Message, SimpleSelect, Textarea } from '@axonivy/ui-components';
+import { getVariable } from '../../../../data/data';
+import type { TreePath } from '../../../../types/config';
 import { metadataOptions, type Variable } from '../../data/Variable';
 
 type VariableProps = {
-  variable?: Variable;
+  variables: Array<Variable>;
+  variablePath: TreePath;
+  setVariables: (variables: Array<Variable>) => void;
 };
 
-export const VariableDetail = ({ variable }: VariableProps) => {
+export const VariableDetail = ({ variables, variablePath, setVariables }: VariableProps) => {
+  const variable = getVariable(variables, variablePath);
   if (!variable) {
     return <Message>Select a variable to edit.</Message>;
   }
 
   const selectedMetadataOption = metadataOptions.find(option => option.value === variable.metadata);
   const doesNotHaveChildren = variable.children.length == 0;
 
+  const handleVariableAttributeChange = (key: keyof Variable, value: any) => {
```

Comment:
make it generic instead of just do any:
`const handleVariableAttributeChange = <TKey extends keyof Variable>(key: TKey, value: Variable[TKey]) => {`

https://www.typescriptlang.org/play?ssl=6&ssc=109&pln=6&pc=3#code/ATAuE8AcFNgNQIYCcCWCBGAbWBeYBvAKBBADsEBbaALmAGdRVSBzAbmJIBNo6BjVSKBQB7UrQZM2HAL4cOvUQ2AALBKU7ZEqDNgCCoRinQBXUNADCqlrmAAeACoBpaOGDQAHmfV1gAaxfCAGbwyGhY0AB8ABT+4LROLgA0wABuCJjGNCHa4QDaCeAAugCUwDgRBLIcVhrQWmF6BqgmZpZqzNBRAETkVF3JAIwATADMxaxAA